src/Compiler/CompilerOptions.cpp

```diff
@@ -1413,11 +1417,11 @@ void initCompilerConfig() {
     setLLVMOption(getLLVMOption() + " --enable-unsafe-fp-math");
   }
 
-  if (march == "z17")
```

Comment:
zXX or archYY are equally recognized, so no need to convert the one into the other